### PR TITLE
Refactored Android JAX-RS

### PR DESCRIPTION
**Pull request by milanperera**
1. Added a function to retrieve list of enrolled devices for a given user.
2. Implemented update pending operations.

### CODE REVIEW

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java [hunk 1]
```diff
@@ -46,26 +46,40 @@
...
-				ArrayList<Operation>();
-		if (log.isDebugEnabled()) {
-			log.debug("Invoking Android pending operations:"+id);
+
+		try {
+			if (resultOperations != null) {
+				updateOperations(resultOperations);
+			}
+		} catch (OperationManagementException e) {
+			message.setResponseMessage("Issue in retrieving operation management service instance");
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java]: Maybe we should assign the error message to a proper String variable and pass it for logging as well as into the instance of "Message".

Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java]: Added separate string for error message

**Comment by dulichan** [comment]
Merged this after locally building

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java [hunk 2]
```diff
@@ -46,26 +46,41 @@
...
+		List<? extends org.wso2.carbon.device.mgt.common.operation.mgt.Operation> operations;
+
 		try {
-			operations = AndroidAPIUtils.getDeviceManagementService().getPendingOperations(deviceIdentifier);
-		}catch(OperationManagementException ex){
-			message.setResponseMessage("Issue in retrieving operation management service instance");
+			operations = AndroidAPIUtils.getPendingOperations(deviceIdentifier);
+		} catch (OperationManagementException e) {
+			String errorMessage = "Issue in retrieving operation management service instance";
+			message.setResponseMessage(errorMessage);
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java]: Looks like we're setting too many attributes upon the instance of "Message". Perhaps, it's good to use "BuilderPattern" or something similar to make the code less bulky.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/LockCode.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/LockCode.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/Notification.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/Notification.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/Operation.java [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/Operation.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/PasswordPolicy.java [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/PasswordPolicy.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/WebClip.java [hunk 7]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/WebClip.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/WebClip.java [hunk 8]
```diff
@@ -1,5 +1,5 @@
...
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/WebClip.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/WipeData.java [hunk 9]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/WipeData.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/BlacklistApplicationsBeanWrapper.java [hunk 10]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/BlacklistApplicationsBeanWrapper.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/BlacklistApplicationsBeanWrapper.java [hunk 11]
```diff
@@ -20,7 +20,7 @@
 import java.util.List;
 
 /**
- * This class is used to wrap the blacklistapplications bean with devices.
+ * This class is used to wrap the BlacklistApplications bean with devices.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/BlacklistApplicationsBeanWrapper.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/CameraBeanWrapper.java [hunk 12]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/CameraBeanWrapper.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/EncryptBeanWrapper.java [hunk 13]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/EncryptBeanWrapper.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/InstallApplicationBeanWrapper.java [hunk 14]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/InstallApplicationBeanWrapper.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/InstallApplicationBeanWrapper.java [hunk 15]
```diff
@@ -20,7 +20,7 @@
 import java.util.List;
 
 /**
- * This class is used to wrap the installapplication bean with devices.
+ * This class is used to wrap the InstallApplication bean with devices.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/InstallApplicationBeanWrapper.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/LockCodeBeanWrapper.java [hunk 16]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/LockCodeBeanWrapper.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/NotificationBeanWrapper.java [hunk 17]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/NotificationBeanWrapper.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/PasswordPolicyBeanWrapper.java [hunk 18]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/PasswordPolicyBeanWrapper.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/UninstallApplicationBeanWrapper.java [hunk 19]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/UninstallApplicationBeanWrapper.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/WebClipBeanWrapper.java [hunk 20]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/WebClipBeanWrapper.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/WifiBeanWrapper.java [hunk 21]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/WifiBeanWrapper.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/WipeDataBeanWrapper.java [hunk 22]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prabathabey) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/WipeDataBeanWrapper.java]: Wrong license header.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java (hunk 2 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java]: Added in PR[https://github.com/wso2/product-mdm/pull/66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/LockCode.java (hunk 3 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/LockCode.java]: Fixed in PR[https://github.com/wso2/product-mdm/pull/66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/Notification.java (hunk 4 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/Notification.java]: Fixed in PR[https://github.com/wso2/product-mdm/pull/66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/Operation.java (hunk 5 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/Operation.java]: Fixed in PR[https://github.com/wso2/product-mdm/pull/66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/PasswordPolicy.java (hunk 6 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/PasswordPolicy.java]: Fixed in PR[https://github.com/wso2/product-mdm/pull/66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/WebClip.java (hunk 7 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/WebClip.java]: Fixed in PR[https://github.com/wso2/product-mdm/pull/66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/WebClip.java (hunk 8 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/WebClip.java]: Fixed in PR[https://github.com/wso2/product-mdm/pull/66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/WipeData.java (hunk 9 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/WipeData.java]: Fixed in PR[https://github.com/wso2/product-mdm/pull/66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/BlacklistApplicationsBeanWrapper.java (hunk 10 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/BlacklistApplicationsBeanWrapper.java]: Fixed in PR[https://github.com/wso2/product-mdm/pull/66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/BlacklistApplicationsBeanWrapper.java (hunk 11 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/BlacklistApplicationsBeanWrapper.java]: Fixed in PR[https://github.com/wso2/product-mdm/pull/66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/CameraBeanWrapper.java (hunk 12 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/CameraBeanWrapper.java]: Fixed in PR[https://github.com/wso2/product-mdm/pull/66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/EncryptBeanWrapper.java (hunk 13 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/EncryptBeanWrapper.java]: Fixed in PR[https://github.com/wso2/product-mdm/pull/66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/InstallApplicationBeanWrapper.java (hunk 14 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/InstallApplicationBeanWrapper.java]: Fixed in PR[https://github.com/wso2/product-mdm/pull/66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/InstallApplicationBeanWrapper.java (hunk 15 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/InstallApplicationBeanWrapper.java]: Fixed in PR[#66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/LockCodeBeanWrapper.java (hunk 16 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/LockCodeBeanWrapper.java]: Fixed in PR[#66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/NotificationBeanWrapper.java (hunk 17 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/NotificationBeanWrapper.java]: Fixed in PR[#66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/PasswordPolicyBeanWrapper.java (hunk 18 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/PasswordPolicyBeanWrapper.java]: Fixed in PR[#66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/UninstallApplicationBeanWrapper.java (hunk 19 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/UninstallApplicationBeanWrapper.java]: Fixed in PR[#66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/WebClipBeanWrapper.java (hunk 20 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/WebClipBeanWrapper.java]: Fixed in PR[#66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/WifiBeanWrapper.java (hunk 21 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/WifiBeanWrapper.java]: Fixed in PR[#66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/WipeDataBeanWrapper.java (hunk 22 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/bean/wrapper/WipeDataBeanWrapper.java]: Fixed in PR[#66]

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java [hunk 23]
```diff
@@ -46,26 +46,41 @@
 
 	private static Log log = LogFactory.getLog(OperationMgtService.class);
 
-	@GET
+	@PUT
 	@Produces(MediaType.APPLICATION_JSON)
 	@Path("{id}")
 	public List<? extends org.wso2.carbon.device.mgt.common.operation.mgt.Operation> getPendingOperations
-			(@HeaderParam("Accept") String acceptHeader, @PathParam("id") String id) {
+			(@HeaderParam("Accept") String acceptHeader, @PathParam("id") String id,
```
Comment (DilanUA) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java]: while the response is built according to the content-type specified by the accept-header, can we just limit the response to JSON by specifying "@Produces(MediaType.APPLICATION_JSON)"?

File: modules/rest-api/src/main/java/org/wso2/carbon/mdm/api/MobileDevice.java [hunk 24]
```diff
@@ -90,4 +90,31 @@
...
+	 * @param user   User Name
+	 * @return Device
+	 * @throws org.wso2.carbon.mdm.api.common.MDMAPIException
+	 *
+	 */
+	@GET
+	@Path("{user}")
+	public List<org.wso2.carbon.device.mgt.common.Device> getDevice(@PathParam("user") String user)
+			throws MDMAPIException {
+		String msg;
```
Comment (DilanUA) [inline modules/rest-api/src/main/java/org/wso2/carbon/mdm/api/MobileDevice.java]: Can declare the msg string within the catch block

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java (hunk 23 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java]: Fixed in PR [#70]

File: modules/rest-api/src/main/java/org/wso2/carbon/mdm/api/MobileDevice.java (hunk 24 above)
Comment (milanperera) [inline modules/rest-api/src/main/java/org/wso2/carbon/mdm/api/MobileDevice.java]: Fixed in PR [#70]

**Pull request merged**